packages/swap/src/routes/quoteV2.ts

```diff
@@ -0,0 +1,365 @@
+import type { Server } from 'socket.io';
+import { findPrice } from '@/amm-addon';
+import {
+  Asset,
+  Assets,
+  Chain,
+  Chains,
+  InternalAsset,
+  getAssetAndChain,
+  getInternalAsset,
+  getInternalAssets,
+} from '@/shared/enums';
+import {
+  bigintMin,
+  getHundredthPipAmountFromAmount,
+  getPipAmountFromAmount,
+} from '@/shared/functions';
+import { ParsedQuoteParams, QuoteQueryResponse, quoteQuerySchema, SwapFee } from '@/shared/schemas';
+import { calculateIncludedSwapFees, estimateIngressEgressFeeAssetAmount } from '@/swap/utils/fees';
+import { estimateSwapDuration } from '@/swap/utils/swap';
+import { asyncHandler } from './common';
+import env from '../config/env';
+import { getPoolsNetworkFeeHundredthPips } from '../consts';
+import { getAssetPrice } from '../pricing';
+import { checkPriceWarning } from '../pricing/checkPriceWarning';
+import getConnectionHandler from '../quoting/getConnectionHandler';
+import { collectMakerQuotes } from '../quoting/quotes';
+import logger from '../utils/logger';
+import { getPools } from '../utils/pools';
+import {
+  getMinimumEgressAmount,
+  getEgressFee,
+  getIngressFee,
+  validateSwapAmount,
+} from '../utils/rpc';
+import ServiceError from '../utils/ServiceError';
+// import { getBrokerQuote } from '../utils/statechain';
+
+const fallbackChains = {
+  [Assets.ETH]: Chains.Ethereum,
+  [Assets.USDC]: Chains.Ethereum,
+  [Assets.FLIP]: Chains.Ethereum,
+  [Assets.BTC]: Chains.Bitcoin,
+  [Assets.DOT]: Chains.Polkadot,
+  [Assets.USDT]: Chains.Ethereum,
+} satisfies Record<Asset, Chain>;
+
+const quote = (io: Server) => {
+  const { handler, quotes$ } = getConnectionHandler();
+
+  io.on('connection', handler);
+
+  const approximateIntermediateOutput = async (asset: InternalAsset, amount: string) => {
+    const price = await getAssetPrice(asset);
+
+    if (typeof price !== 'number') return null;
+
+    return BigInt(new BigNumber(amount).times(price).toFixed(0));
+  };
+
+  async function quoteLeg(
+    [baseAsset, quoteAsset]: readonly [InternalAsset, InternalAsset],
+    amount: null,
+  ): Promise<null>;
+  async function quoteLeg(
+    [baseAsset, quoteAsset]: readonly [InternalAsset, InternalAsset],
+    amount: bigint,
+  ): Promise<bigint>;
+  async function quoteLeg(
+    [baseAsset, quoteAsset]: readonly [InternalAsset, InternalAsset],
+    amount: bigint | null,
+  ): Promise<bigint | null>;
+  async function quoteLeg(
+    [baseAsset, quoteAsset]: readonly [InternalAsset, InternalAsset],
+    amount: bigint | null,
+  ): Promise<bigint | null> {
+    if (!amount) return null;
+
+    const id = randomUUID();
+
+    io.emit('quote_request', {
+      request_id: id,
+      amount: String(amount),
+      base_asset: getAssetAndChain(baseAsset),
+      quote_asset: getAssetAndChain(quoteAsset),
+    });
+
+    const quotes = await collectMakerQuotes(id, io.sockets.sockets.size, quotes$);
+
+    const { swappedAmount, remainingAmount } = await findPrice({
```

Comment:
is this the only place where the amm addon is called?

Reply:
yes